File: python/SelectionSort/src/SelectionSort.py
# Python program for implementation of Insertion Sort
import sys

# main
list = map(int, sys.argv[1:len(sys.argv)])
print "Input list::", list

# Function to do insertion sort
def selectionSort(list):

    # Traverse through 1 to len(list)
    for i in range(len(list)):

        minIndex = i

        print "minIndex::", i

        for j in range(i+1, len(list)):
            if list[minIndex] > list[j]:
                minIndex = j

        # Swap the found minimum element with
        # the first element
        list[i], list[minIndex] = list[minIndex], list[i]

        print "swap::[", list[minIndex], " <--> ", list[i], "] position::[", i, " <--> ", minIndex, "]"
        print "Partial list::", list

# Driver code to test above
selectionSort(list)
print "Output list::", list

File: java/SelectionSort/src/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {

  public static void main(String[] list) {

    int minValue;

    System.out.println("Input::".concat(Arrays.toString(list)));

    for(int i = 0; i < list.length; i++) {

      minValue = i;

      System.out.println("iteration::" + i);

      for (int j = i; j < list.length; j++) {
        if( Integer.parseInt(list[minValue]) > Integer.parseInt(list[j])) {
          minValue = j;
        }
      }

      swap(list, i, minValue);

      System.out.println("Partial::".concat(Arrays.toString(list)));
    }

    System.out.println("Output:::".concat(Arrays.toString(list)));
  }

  //do swap
  private static void swap(String[] list, int from, int to) {
    System.out.println("swap::[" + list[from] + " <--> " + list[to] + "] position::["+ from + " <--> " + to + "]");
    String temp = list[to];
    list[to] = list[from];
    list[from] = temp;
  }
}
